gensim/similarities/termsim.py

```diff
@@ -149,44 +161,53 @@ class SparseTermSimilarityMatrix(SaveLoad):
+        of the resulting term similarity matrix. The dictionary may only be None when source is
         a :class:`scipy.sparse.spmatrix`.
     tfidf : :class:`gensim.models.tfidfmodel.TfidfModel` or None, optional
         A model that specifies the relative importance of the terms in the dictionary. The columns
         of the term similarity matrix will be build in a decreasing order of importance of
         terms, or in the order of term identifiers if None.
     symmetric : bool, optional
-        Whether the symmetry of the term similarity matrix will be enforced. This parameter only has
-        an effect when `source` is a :class:`scipy.sparse.spmatrix`. Positive definiteness is a
-        necessary precondition if you later wish to derive a change-of-basis matrix from the term
-        similarity matrix using Cholesky factorization.
-    positive_definite: bool, optional
-        Whether the positive definiteness of the term similarity matrix will be enforced through
-        strict column diagonal dominance. Positive definiteness is a necessary precondition if you
-        later wish to derive a change-of-basis matrix from the term similarity matrix using Cholesky
-        factorization.
+        Whether the symmetry of the term similarity matrix will be enforced. Symmetry is a necessary
+        precondition for positive definiteness, which is necessary if you later wish to derive a
+        unique change-of-basis matrix from the term similarity matrix using Cholesky factorization.
+        Setting symmetric to False will significantly reduce memory usage during matrix construction.
+    dominant: bool, optional
+        Whether the strict column diagonal dominance of the term similarity matrix will be enforced.
+        Strict diagonal dominance and symmetry are sufficient preconditions for positive
+        definiteness, which is necessary if you later wish to derive a change-of-basis matrix from
+        the term similarity matrix using Cholesky factorization.
     nonzero_limit : int or None, optional
         The maximum number of non-zero elements outside the diagonal in a single column of the
         sparse term similarity matrix. If None, then no limit will be imposed.
     dtype : numpy.dtype, optional
-        Data-type of the sparse term similarity matrix.
+        The data type of the sparse term similarity matrix.
+    positive_definite: bool or None, optional
+        A deprecated alias for dominant.
 
     Attributes
     ----------
     matrix : :class:`scipy.sparse.csc_matrix`
         The encapsulated sparse term similarity matrix.
-    """
-    PROGRESS_MESSAGE_PERIOD = 1000  # how many columns are processed between progress messages
 
-    def __init__(self, source, dictionary=None, tfidf=None, symmetric=True, positive_definite=False, nonzero_limit=100,
-                 dtype=np.float32):
+    """
+    def __init__(self, source, dictionary=None, tfidf=None, symmetric=True, dominant=False,
```

Comment:
This initializer does a lot of work... I think it would be helpful to reduce it. On top of that, the whole pattern of re-calling the initializer from itself is ugly. I think this is a good opportunity to rework it.

Start with this:

```python
def _create_source(index, dictionary=None, tfidf=None, symmetric=True, ...):
    # do the heavy lifting here
    return matrix 
```

and then simplify the initializer to:

```python
# deal with the old positive_definite arg here and then forget about it

if not sparse.issparse(source):
    source = _create_source(source, dictionary=dictionary, symmetric=symmetric, ...)
self.matrix = source.tocsc()
```

Reply:
Fixed in 621b2525631b4146f4a301f1f84e47500869fdf0.